server/src/services/asset.service.ts

```diff
@@ -252,11 +252,19 @@ export class AssetService {
   async update(auth: AuthDto, id: string, dto: UpdateAssetDto): Promise<AssetResponseDto> {
     await this.access.requirePermission(auth, Permission.ASSET_UPDATE, id);
 
+    let asset = await this.assetRepository.getById(id);
```

Comment:
this is going to cause a second query right after having done a query on the line above. Perhaps it should just be part of that initial permission check. see `AssetAcess.checkOwnerAccess` in `access.repository.ts`

Reply:
That would be fine with me. Asset update access implies the asset is not read only.